### PR TITLE
chore: update readme and release from CI

### PR DESCRIPTION
**Pull request by pratik151192**
Added dependency info to README now that we have a release out, and also changed CI to not just upload artifacts to Sonatype but also release it automatically now that we have verified it.

### CODE REVIEW

File: .github/workflows/manual_release.yml [hunk 1]
```diff
@@ -47,4 +47,4 @@ jobs:
           MOMENTO_DDB_LOCK_VERSION: ${{ steps.semrel.outputs.version }}
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: publishToSonatype closeSonatypeStagingRepository
+          arguments: publishToSonatype closeAndReleaseStagingRepository
```
Comment (malandis) [inline .github/workflows/manual_release.yml:50]: What does this do?

**Review by malandis — COMMENTED** [review]
1 q

File: .github/workflows/manual_release.yml (hunk 1 above)
Comment (pratik151192) [reply .github/workflows/manual_release.yml:50]: So `closeSonatypeStagingRepository` simply uploads to the repository and doesn't release. That's just for sanity check - and the screenshot that I shared on Slack. This wouldn't require us to manually release each type, and Nexus will automatically release it for us. This is how the Java SDK is also wired.

Comment (pratik151192) [reply .github/workflows/manual_release.yml:50]: Essentially the current one on `main` just uploads artifacts and requires a manual release, and the one in the PR is automated to do releases through CI action.

**Review by malandis — APPROVED** [review]

**Pull request merged**